src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -113,12 +126,68 @@ private static bool IsValidForGenericConstraintContext(ISymbol symbol)
 
             // If current symbol is a struct or static or sealed class then it cannot be used as a generic constraint.
             // However it can contain other valid constraint types and if this is true we should show it
-            if (namedType.IsStructType() || namedType.IsStatic || namedType.IsSealed)
+            if (namedType.IsModuleType() || namedType.IsStructType() || namedType.IsStatic || namedType.IsSealed)
+                return namedType.GetTypeMembers().Any(static m => IsValidForGenericConstraintContext(m));
+
+            return true;
+        }
+
+        private static bool IsValidForInheritanceContext(ISymbol symbol, INamedTypeSymbol? inheritingFrom, TSyntaxContext context)
+        {
+            if (symbol is IAliasSymbol alias)
+                symbol = alias.Target;
+
+            if (symbol is INamespaceSymbol @namespace)
+                return @namespace.GetMembers().Any(m => IsValidForInheritanceContext(m, inheritingFrom, context));
```

Comment:
please don't want down namespaces.  they can be enormous and this is effectively a large graph walk.  it seems reasonable to presume that most namespaces would have at least some type (unsealed class/interface/etc.) that would make them viable.

Reply:
What to do with this fellow then?)
https://github.com/dotnet/roslyn/blob/e2c555e116bb7cb950f82c25430f6aa466c8a33d/src/EditorFeatures/VisualBasicTest/Completion/CompletionProviders/SymbolCompletionProviderTests.vb#L5491-L5505
I am ok to just delete it, but I need your confirmation to do that

Reply:
i'm ok changing that test.